src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const prop in action.extraData) {
+          state[prop] = action.extraData[prop];
+        }
+        break;
+      case 'clear':
+        for (const stateProp in state) {
+          delete state[stateProp];
+        }
+        break;
+      case 'removeProperties':
+        for (const prop of action.keysToRemove) {
+          delete state[prop];
+        }
+        break;
+    }
+  }
+
+  return state;
```

Comment:
you don't need to return state in this task, all you need is to change it, so this return statement is redundant
```suggestion
```